sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQueryStatusAndProgressSuite.scala

```diff
@@ -38,13 +38,18 @@ class StreamingQueryStatusAndProgressSuite extends SparkFunSuite {
         |  "id" : "${testProgress1.id.toString}",
         |  "runId" : "${testProgress1.runId.toString}",
         |  "name" : "myName",
-        |  "timestamp" : "2016-12-05T20:54:20.827Z",
+        |  "triggerTimestamp" : "2016-12-05T20:54:20.827Z",
         |  "numInputRows" : 678,
         |  "inputRowsPerSecond" : 10.0,
         |  "durationMs" : {
         |    "total" : 0
         |  },
-        |  "currentWatermark" : 3,
+        |  "queryTimestamps" : {
+        |    "eventTime.avg" : "2016-12-05T20:54:20.827Z",
```

Comment:
One option is we could use dashes instead, but honestly I'm getting a little confused by all these timestamps.  In particular, I'm not sure what is the difference between `triggerTimestamp` and `processingTime` (and, I think that having `processingTime` mean anything different than `System.currentTimeMillis` will be confusing to users coming from other systems).

The two things that I think you really want to be able to track from the metrics feed are:
 - the actual timestamp when this progress update was produced.  I think this should remain top level and be called `timestamp`.
 - Stats about the event time, so that I can know roughly what data is present in this batch.  This can be useful for several reasons, including finding other problems upstream.  The more I think about this, I think you just want to see this as `"eventTime": { "min": ..., "max": ..., "watermark": ... }`

Its actually not clear to me why you need to know the original batch start time.  In 99% of cases this is the same as the `timestamp`.  If you are executing, a batch due to a failure, it will be different.  But, why does an outside monitoring job care?  I can't come up with any interesting graphs that you would make with this extra field.